### PR TITLE
View copy patch

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes fusion guard build of view, which triggers assert failures in `graph_fuser.cpp`

```
        TORCH_INTERNAL_ASSERT(                                                                                              
            view->kind() == prim::view_copy ||                                                                                                                                                                     
            view->kind() == prim::reshape_copy);
```

The old logic to find view_copy as a consumer of profiled constants are incorrect. We get lucky in our unit tests since we have two inputs both going to view_copy (tensor and shape) so the problem never arises.

### CODE REVIEW

File: test/test_jit_cuda_fuser.py [hunk 1]
```diff
@@ -3812,6 +3812,23 @@ def run(fn):
...
+    @unittest.skipIf(not RUN_CUDA, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_view_copy_graph_guard(self):
+        x = torch.randn(4, 2, 3, device="cuda").permute([1, 2, 0])
+        y = [4, 6]
+
+        with nvfuser_singleton_fusion(True):
+            def t(x, y : List[int]):
+                t1 = t0 + 1.0
```
Comment (rdspring1) [inline test/test_jit_cuda_fuser.py]: `t0 -> x`

File: test/test_jit_cuda_fuser.py [hunk 2]
```diff
@@ -3812,6 +3812,23 @@ def run(fn):
...
+                t2 = t1 * 1.0
+                out = t2.view(y)
```
Comment (rdspring1) [inline test/test_jit_cuda_fuser.py]: `view -> reshape`

**Review by rdspring1 — APPROVED** [review]
LGTM

**Comment by jjsjann123** [comment]
Tests passed, merging

**Pull request merged**